algebird-core/src/main/scala/com/twitter/algebird/QTree.scala

```diff
@@ -367,3 +369,29 @@ case class QTree[A](
     (n.toDouble(ll) / luc, n.toDouble(uu) / ulc)
   }
 }
+
+trait QTreeAggregatorLike[T] {
+  val percentile: Double
+  val k: Int
+  implicit val num: Numeric[T]
+  def prepare(input: T) = QTree(num.toDouble(input))
+  val semigroup = new QTreeSemigroup[Double](k)
```

Comment:
why don't we want `QTreeSemigroup[Unit]` here? That will be less memory and won't hurt our ability to do percentiles.